data/virt_autotest/setup_dns_service.sh

```diff
@@ -174,48 +229,153 @@ for single_nameserver in ${nameservers_array[@]};do
 done	
 sed -irn "s/forwarders.*$/forwarders { `echo -e ${forwarders_array[@]}` };/" ${dns_config_file}
+#Insert script signature to the end of ${dns_config_file} and ${dns_resolv_file}
+echo -e "#$0" >> ${dns_config_file}
+echo -e "#$0" >> ${dns_resolv_file}
 
 #Start named service. Quit if failed.
 dns_service_name="named"
 get_os_installed_release=`lsb_release -r | grep -oE "[[:digit:]]{2}"`
 os_installed_release=$(echo ${get_os_installed_release})
 if [[ ${os_installed_release} -gt '11' ]];then
-	systemctl enable ${dns_service_name}
-	systemctl restart ${dns_service_name}
-	dns_service_failed=$(echo $?)
-	systemctl status ${dns_service_name} | tee -a ${setup_log_file}
+        systemctl enable ${dns_service_name}
+        systemctl restart ${dns_service_name}
+        dns_service_failed=$(echo $?)
+        systemctl status ${dns_service_name} | tee -a ${setup_log_file}
 else
-	service ${dns_service_name} restart
-	dns_service_failed=$(echo $?)
-	service ${dns_service_name} status | tee -a ${setup_log_file}
+        service ${dns_service_name} restart
+        dns_service_failed=$(echo $?)
+        service ${dns_service_name} status | tee -a ${setup_log_file}
 fi
 if [[ ${dns_service_failed} -ne 0 ]];then
-	echo -e "DNS service did not start up as normal. Please investigate.\n" | tee -a ${setup_log_file}
-	echo -e "DNS resolver file content:\n" | tee -a ${setup_log_file}
-	cat ${dns_resolv_file} | tee -a ${setup_log_file}
-	echo -e "NAMED service file content:\n" | tee -a ${setup_log_file}
-	cat ${dns_config_file} | tee -a ${setup_log_file}
-	exit 1
+        echo -e "DNS service did not start up as normal. Please investigate.\n" | tee -a ${setup_log_file}
+        echo -e "DNS resolver file content:\n" | tee -a ${setup_log_file}
+        cat ${dns_resolv_file} | tee -a ${setup_log_file}
+        echo -e "NAMED service file content:\n" | tee -a ${setup_log_file}
+        cat ${dns_config_file} | tee -a ${setup_log_file}
+        exit 1
 fi
 
-#Ping vm machines one by one. Quit if any failure.
-sleep 45s
+#Generate ssh key and set it as IdentityFile in /etc/ssh/ssh_config
+ssh_key_pass="novell"
+ssh_key_path="/var/${dns_domain_forward}/.ssh"
+ssh_config_file="/etc/ssh/ssh_config"
+ssh_config_temp="/etc/ssh/ssh_config.temp"
+cat ${ssh_config_file} | grep "#$0"
+if [ `echo $?` == 0 ];then
+        mv ${ssh_config_file}.orig ${ssh_config_file}
+fi
+cp ${ssh_config_file} ${ssh_config_file}.orig
+if [ -e ${ssh_key_path}/id_rsa ];then
+        rm -f ${ssh_key_path}/id_rsa
+        rm -f ${ssh_key_path}/id_rsa.pub
+fi
+rm -f -r ${ssh_key_path}
+mkdir -p ${ssh_key_path}
+ssh-keygen -t rsa -f ${ssh_key_path}/id_rsa -q -P "" | tee -a ${setup_log_file}
+awk -v sshvar="${ssh_key_path}/id_rsa" 'done != 1 && /^#.*IdentityFile.*$/ { print "IdentityFile "sshvar"\n"; done=1 } 1' ${ssh_config_file} > ${ssh_config_temp}
+mv ${ssh_config_temp} ${ssh_config_file}
+#Insert script signature to the end of ${ssh_config_file}
+echo -e "#$0" >> ${ssh_config_file}
+
```

Comment:
I guess the above section for ssh keys is to avoid password typing during dns visit? If yes, would you please add some comment describing the purpose and how you do it?